commit 85a088cc4c8eeeb2e4cf6c7e0efde52045360b05
Author: Takashi Iwai <tiwai@suse.de>
Date:   Wed Sep 12 12:58:43 2018 +0200

    drm/nouveau: Don't disable polling in fallback mode
    
    [ Upstream commit 118780066e30c34de3d9349710b51780bfa0ba83 ]
    
    When a fan is controlled via linear fallback without cstate, we
    shouldn't stop polling.  Otherwise it won't be adjusted again and
    keeps running at an initial crazy pace.
    
    Fixes: 800efb4c2857 ("drm/nouveau/drm/therm/fan: add a fallback if no fan control is specified in the vbios")
    Bugzilla: https://bugzilla.suse.com/show_bug.cgi?id=1103356
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=107447
    Reported-by: Thomas Blume <thomas.blume@suse.com>
    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Reviewed-by: Martin Peres <martin.peres@free.fr>
    Signed-off-by: Ben Skeggs <bskeggs@redhat.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/gpu/drm/nouveau/nvkm/subdev/therm/base.c b/drivers/gpu/drm/nouveau/nvkm/subdev/therm/base.c
index 3695cde669f8..07914e36939e 100644
--- a/drivers/gpu/drm/nouveau/nvkm/subdev/therm/base.c
+++ b/drivers/gpu/drm/nouveau/nvkm/subdev/therm/base.c
@@ -132,11 +132,12 @@ nvkm_therm_update(struct nvkm_therm *therm, int mode)
 			duty = nvkm_therm_update_linear(therm);
 			break;
 		case NVBIOS_THERM_FAN_OTHER:
-			if (therm->cstate)
+			if (therm->cstate) {
 				duty = therm->cstate;
-			else
+				poll = false;
+			} else {
 				duty = nvkm_therm_update_linear_fallback(therm);
-			poll = false;
+			}
 			break;
 		}
 		immd = false;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1399_85a088cc4c8eeeb2e4cf6c7e0efde52045360b05.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1399_85a088cc4c8eeeb2e4cf6c7e0efde52045360b05.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
